homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -0,0 +1,514 @@
+            start_server_info = await self.hass.async_add_job(
+                self.leaf.get_latest_battery_status
+            )
+
+            # Store the date from the nissan servers
+            start_date = self._extract_start_date(start_server_info)
+            if start_date is None:
+                _LOGGER.info("No start date from servers. Aborting")
+                return None
+
+            _LOGGER.info("Start server date=%s", start_date)
+
+            # Request battery update from the car
+            _LOGGER.info("Requesting battery update, %s", self.leaf.vin)
+            request = await self.hass.async_add_job(self.leaf.request_update)
+            if not request:
+                _LOGGER.error("Battery update request failed")
+                return None
+
+            for attempt in range(MAX_RESPONSE_ATTEMPTS):
+                _LOGGER.info("Waiting %s seconds for battery update (%s) (%s)",
+                             PYCARWINGS2_SLEEP, self.leaf.vin, attempt)
+                await asyncio.sleep(PYCARWINGS2_SLEEP)
+
+                # Note leaf.get_status_from_update is always returning 0, so
+                # don't try to use it anymore.
+                server_info = await self.hass.async_add_job(
+                    self.leaf.get_latest_battery_status
+                )
+
+                latest_date = self._extract_start_date(server_info)
+                _LOGGER.info("Latest server date=%s", latest_date)
+                if latest_date is not None and latest_date != start_date:
+                    return server_info
+
+            _LOGGER.info("%s attempts exceeded return latest data from server",
+                         MAX_RESPONSE_ATTEMPTS)
+            return server_info
+        except CarwingsError:
+            _LOGGER.error("An error occurred getting battery status.")
+            return None
+
+    async def async_get_climate(self):
+        """Request climate data from Nissan servers."""
+        from pycarwings2 import CarwingsError
+        try:
+            request = await self.hass.async_add_job(
+                self.leaf.get_latest_hvac_status
+            )
+            return request
+        except CarwingsError:
+            _LOGGER.error(
+                "An error occurred communicating with the car %s",
+                self.leaf.vin)
+            return None
+
+    async def async_set_climate(self, toggle):
+        """Set climate control mode via Nissan servers."""
+        climate_result = None
+        if toggle:
+            _LOGGER.info("Requesting climate turn on for %s", self.leaf.vin)
+            request = await self.hass.async_add_job(
+                self.leaf.start_climate_control
+            )
+            for attempt in range(MAX_RESPONSE_ATTEMPTS):
+                if attempt > 0:
+                    _LOGGER.info("Climate data not in yet (%s) (%s). "
+                                 "Waiting (%s) seconds.", self.leaf.vin,
```

Comment:
We don't end logging messages with period.

Reply:
Fixed in commit https://github.com/filcole/home-assistant/commit/62cf3ea463b36bdd7dd25e5865c80b7b9d9dade9